ctapipe/io/hdf5tableio.py

```diff
@@ -505,9 +509,20 @@ def _map_table_to_containers(
                     )
 
             # store the meta
-            self._meta[table_name] = {}
-            for key in tab.attrs._f_list():
-                self._meta[table_name][key] = tab.attrs[key]
+            self._meta[table_name][container.__name__] = {}
+            for key in container.fields:
```

Comment:
Wouldn't it make more sense to loop through the available meta and sort that into the containers instead of looking for all possible combinations?

Reply:
Probably also would be good to break this out into a `handle_metadata` function

Reply:
Either way I need to strip the (possible) prefix and suffix from the meta key in order to correctly assign those values. I found it easier to create the pre-and-suffixed version from the plain one instead of the other way around (stripping prefix and suffix).